### PR TITLE
Docs: Add Japanese Astro vs. X page

### PR DESCRIPTION
**Pull request by kyosuke**
## Changes

- Translated the Astro vs. X page into Japanese.

## Testing

I have confirmed that all the links work in my local environment.

## Docs

Document update only

### CODE REVIEW

File: docs/src/pages/ja/comparing-astro-vs-other-tools.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Astro vs. X
+description: Astroと他の静的サイトジェネレーター（Gatsby、Next.js。Nuxt、Hugo、Eleventyなど）の比較
```
Comment (togami2864) [inline docs/src/pages/ja/comparing-astro-vs-other-tools.md:4]: typo
`Next.js。` -> `Next.js、`

**Review by togami2864 — COMMENTED** [review]
This is already looking quite good with one small nit !

File: docs/src/pages/ja/comparing-astro-vs-other-tools.md (hunk 1 above)
Comment (kyosuke) [reply docs/src/pages/ja/comparing-astro-vs-other-tools.md]: Thanks! I fixed the typo.

**Review by togami2864 — COMMENTED** [review]
LGTM🎉

**Review by drwpow — APPROVED** [review]

**Pull request merged**